INSTALL.rst

```diff
@@ -0,0 +1,436 @@
+Building cvc5
+-------------
+
+.. code:: bash
+
+    ./configure.sh
+        # use --prefix to specify an install prefix (default: /usr/local)
+        # use --name=<PATH> for custom build directory
+        # use --auto-download to download and build missing, required or
+        #   enabled, dependencies
+    cd <build_dir>   # default is ./build
+    make             # use -jN for parallel build with N threads
+    make check       # to run default set of tests
+    make install     # to install into the prefix specified above
+
+All binaries are built into ``<build_dir>/bin``, the cvc5 library is built into
+``<build_dir>/lib``.
+
+
+Supported Operating Systems
+---------------------------
+
+cvc5 can be built natively on Linux and macOS, cross-compilation is possible for
+Windows using Mingw-w64. cvc5 also supports cross-compilation for ARM64 systems.
+We generally recommend a 64-bit operating system.
+
+
+Compilation on macOS
+^^^^^^^^^^^^^^^^^^^^
+
+On macOS, we recommend using `Homebrew <https://brew.sh/>`_ to install the
+dependencies.  We also have a Homebrew Tap available at
+https://github.com/CVC4/homebrew-cvc4 .
+To build a static binary for macOS, use:
+``./configure.sh --static --no-static-binary``.
+
+
+Cross-compiling for Windows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cross-compiling cvc5 with Mingw-w64 can be done as follows:
+
+.. code:: bash
+
+  ./configure.sh --win64 --static <configure options...>
+
+  cd <build_dir>   # default is ./build
+  make             # use -jN for parallel build with N threads
+
+The built binary ``cvc5.exe`` is located in ``<build_dir>/bin`` and the cvc5
+library can be found in ``<build_dir>/lib``.
+
+
+Build dependencies
+------------------
+
+cvc5 makes uses of a number of tools and libraries. Some of these are required
+while others are only used with certain configuration options. If
+``--auto-download`` is given, cvc5 can automatically download and build most
+libraries that are not already installed on your system. If your libraries are
+installed in a non-standard location, you can use ``--dep-path`` to define an
+additional search path for all dependencies. Versions given are minimum
+versions; more recent versions should be compatible.
+
+- `GNU C and C++ (gcc and g++) <https://gcc.gnu.org>`_
+  or `Clang <https://clang.llvm.org>`_ (reasonably recent versions)
```

Comment:
Instead of "reasonably recent versions", could we say that we require "GCC >= 7 and Clang >= 5"?